tests/helpers.py

```diff
@@ -87,6 +87,77 @@ def convolve(_input, filter_weights, filter_bias, strides, filter_sizes,
     return _input
 
 
+def recurrent_step(input_val,
+                   num_units,
+                   step_hidden,
+                   step_cell,
+                   w_x_init,
+                   w_h_init,
+                   b_init,
+                   nonlinearity,
+                   gate_nonlinearity,
+                   forget_bias=1.0):
+    """
+    A LSTM unit implements the following update mechanism:
+
+    Incoming gate:    i(t) = f_i(x(t) @ W_xi + h(t-1) @ W_hi +
+                                 w_ci * c(t-1) + b_i)
+    Forget gate:      f(t) = f_f(x(t) @ W_xf + h(t-1) @ W_hf +
+                                 w_cf * c(t-1) + b_f)
+    Cell gate:        c(t) = f(t) * c(t - 1) + i(t) * f_c(x(t) @ W_xc +
+                             h(t-1) @ W_hc + b_c)
+    Out gate:         o(t) = f_o(x(t) @ W_xo + h(t-1) W_ho + w_co * c(t) + b_o)
+    New hidden state: h(t) = o(t) * f_h(c(t))
+
+    Note that the incoming, forget, cell, and out vectors must have the same
+    dimension as the hidden state.
+    """
+
+    def f(x):
+        return x
+
+    if nonlinearity is None:
+        nonlinearity = f
+    if gate_nonlinearity is None:
+        gate_nonlinearity = f
+
+    input_dim = np.prod(input_val.shape[1:])
+    # Weights for the input gate
+    w_xi = np.full((input_dim, num_units), w_x_init)
+    w_hi = np.full((num_units, num_units), w_h_init)
+    b_i = np.full((num_units, ), b_init)
+    # Weights for the forget gate
+    w_xf = np.full((input_dim, num_units), w_x_init)
+    w_hf = np.full((num_units, num_units), w_h_init)
+    b_f = np.full((num_units, ), b_init)
+    # Weights for the cell gate
+    w_xc = np.full((input_dim, num_units), w_x_init)
+    w_hc = np.full((num_units, num_units), w_h_init)
+    b_c = np.full((num_units, ), b_init)
+    # Weights for the out gate
+    w_xo = np.full((input_dim, num_units), w_x_init)
+    w_ho = np.full((num_units, num_units), w_h_init)
+    b_o = np.full((num_units, ), b_init)
+
+    w_x_ifco = np.concatenate([w_xi, w_xf, w_xc, w_xo], axis=1)
+    w_h_ifco = np.concatenate([w_hi, w_hf, w_hc, w_ho], axis=1)
+
+    x_ifco = np.matmul(input_val, w_x_ifco)
+    h_ifco = np.matmul(step_hidden, w_h_ifco)
+
+    x_i, x_f, x_c, x_o = np.split(x_ifco, 4, axis=1)  # noqa: E501, pylint: disable=unbalanced-tuple-unpacking
```

Comment:
what is the justifcation for disabling E501 and the pylint warning?

Reply:
Since I can't find out what's the reason pylint is complaining about that error, so I disable the pylint checking and since it is too long, I disabled E501 as well. I will dig into it and see what pylint is talking about.

Reply:
perhaps is one of your return values itself a tuple or list?

Reply:
no... They are all `numpy.ndarray`.

Reply:
Could be because interpreter cannot ensure the right hand side can be split into 4. Is there any other way to pass pylint for this?

Reply:
generally pylint rules try to encourage being very explicit and writing code so that it can be statically analyzed for errors (e.g. using pylint).

in this case i think pylint has good point, because the `np.split` operation is totally static (all inputs are known at compile time), but it has been written as a dynamic operation unnecessarily.

the rationale for this is that code is read much more often than it is written, so it's usually better to make it longer and easier to read rather than shorted and easier to write.

maybe it would be clearer if you just wrote out. this way a reader doesn't have to look up documentation for np.split to find out how your array is unpacked.

```python
x_i, x_f, x_c, x_o = x_ifco[:, :4], x_ifco[:, 4:8], x_ifco[:, 8:12], x_ifco[:, 12:16]
# or
x_i = x_ifco[:, :4]
x_f = x_ifco[:, 4:8]
x_c = x_ifco[:, 8:12]
x_o = x_ifco[:, 12:16]
```

Reply:
Got it.